### PR TITLE
AAV-846: Engine integration

### PR DESCRIPTION
**Pull request by alexey-tatarenko**
At the moment manually created solution and CMake-based project both are supported. However Github YAML-script uses CMake to build the project and CTest to perform unit testing. We have to choose one of these approaches ASAP to go further.

### CODE REVIEW

File: .github/workflows/7zip-build.yml [hunk 1]
```diff
@@ -0,0 +1,93 @@
+name: "[Engine] Build"
```
Comment (MaksymSlota) [inline .github/workflows/7zip-build.yml]: name: "Build 7zip DLL"

File: .github/workflows/7zip-build.yml [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+on:
+  push:
```
Comment (MaksymSlota) [inline .github/workflows/7zip-build.yml]: workflow_dispatch:
instead of `push:`

File: .github/workflows/7zip-build.yml [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+env:
+  BUILD_CONFIGURATION: Release
+  DEPENDENCIES_DIR: ${{ github.workspace }}\build\dependencies
+  ARTIFACTS_ROOT_DIR: ${{ github.workspace }}\build\artifacts
+  GDRIVE_DEPENDENCIES_DIR: packages:/7zip/dependencies
```
Comment (MaksymSlota) [inline .github/workflows/7zip-build.yml]: let's move path `packages:/7zip/dependencies`
 to secrtets

File: .github/workflows/7zip-build.yml [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+
+jobs:
+  build_7zip:
+    name: Build 7zip DLL
+    runs-on: windows-2022
+
+    strategy:
```
Comment (MaksymSlota) [inline .github/workflows/7zip-build.yml]: why do you need it?

File: CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,285 @@
+cmake_minimum_required(VERSION 3.14)
```
Comment (MaksymSlota) [inline CMakeLists.txt:1]: why do we need cmake?

File: Scripts/build.ps1 [hunk 6]
```diff
@@ -0,0 +1,13 @@
+param (
+    [Parameter(Mandatory)][string] $sln,
+    [Parameter(Mandatory)][string] $configuration,
+    [Parameter(Mandatory)][string] $platform,
+    [string] $buildNumber = "0000"
+)
+
+. $PSScriptRoot\utils.ps1
+
+Write-Host "Building $sln $confgiguration $platform..."
```
Comment (MaksymSlota) [inline Scripts/build.ps1:10]: 7zip.dll doesn't have a version number in metadata for some reason, please check it...

File: ThirdParty/catch2/catch_reporter_teamcity.hpp [hunk 7]
```diff
@@ -0,0 +1,219 @@
+/*
```
Comment (MaksymSlota) [inline ThirdParty/catch2/catch_reporter_teamcity.hpp]: why do we need it?

File: .github/workflows/7zip-build.yml (hunk 1 above)
Comment (alexey-tatarenko) [reply .github/workflows/7zip-build.yml]: Fixed.

File: ThirdParty/catch2/catch_reporter_tap.hpp [hunk 8]
```diff
@@ -0,0 +1,254 @@
+/*
```
Comment (MaksymSlota) [inline ThirdParty/catch2/catch_reporter_tap.hpp]: why do we need it?

File: ThirdParty/catch2/catch_reporter_sonarqube.hpp [hunk 9]
```diff
@@ -0,0 +1,181 @@
+/*
```
Comment (MaksymSlota) [inline ThirdParty/catch2/catch_reporter_sonarqube.hpp]: why do we need it?

File: .github/workflows/7zip-build.yml [hunk 10]
```diff
@@ -0,0 +1,93 @@
...
+
+        $VersionFile = Get-Content ${{ github.workspace }}\C\7zVersion.h -Raw
+        $RegexProductVersion = 'MY_VERSION_NUMBERS[\s=]+"(?<version>[\d\.]+)"'
+
+        if ($VersionFile -match $RegexProductVersion) {
+          $match = $Matches.version
+          echo "PRODUCT_VERSION=$match" | Out-File -FilePath $env:GITHUB_ENV -Append
+        }
+
+    # Build and test 7zip
```
Comment (MaksymSlota) [inline .github/workflows/7zip-build.yml]: build and test should be different steps

File: .github/workflows/7zip-build.yml [hunk 11]
```diff
@@ -0,0 +1,93 @@
...
+      uses: AnimMouse/setup-rclone@v1
+      with:
+        rclone_config: ${{ secrets.RCLONE_CONFIG }}
+
+    # Set env variables
+    - name: Set environment variables
+      shell: powershell
+      run: |
+        $BRANCH = "${{ github.ref_name }}"
+        $RELEASE = $BRANCH.StartsWith('rev/')
```
Comment (MaksymSlota) [inline .github/workflows/7zip-build.yml]: lets keep it in a separate PowerShell script.
e.g. `getVersion.ps1`

**Review by MaksymSlota — APPROVED** [review]

**Pull request merged**